### PR TITLE
Add function node in azure explorer

### PR DESCRIPTION
**Pull request by Flanker32**
Add function node in azure explorer

### CODE REVIEW

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.microsoft.azuretools.telemetry.TelemetryConstants.*;
+
+public class FunctionNode extends WebAppBaseNode implements FunctionNodeView {
+    private final static Logger LOGGER = Logger.getLogger(FunctionNode.class.getName());
+    private static final String DELETE_FUNCTION_PROMPT_MESSAGE = "This operation will delete the Function App: %s.\n"
+            + "Are you sure you want to continue?";
```
Comment (andxu) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java]: Are you sure to continue

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    private static final String DELETE_FUNCTION_PROGRESS_MESSAGE = "Deleting Function App";
+    private static final String LABEL = "Function";
```
Comment (andxu) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java]: FUNCTION_LABEL

Comment (Flanker32) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNode.java]: Renamed

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNodePresenter.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import com.microsoft.tooling.msservices.serviceexplorer.azure.webapp.base.WebAppBaseState;
+
+import java.io.IOException;
+
+public class FunctionNodePresenter<V extends FunctionNodeView> extends MvpPresenter<V> {
+    public void onStartFunctionApp(String subscriptionId, String appId) throws IOException {
+        AzureFunctionMvpModel.getInstance().startFunction(subscriptionId, appId);
+        final FunctionNodeView view = getMvpView();
+        if (view != null) {
+            view.renderNode(WebAppBaseState.RUNNING);
```
Comment (andxu) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNodePresenter.java]: Can we just trigger a fresh for given appId, other than hard-coding WebAppBaseState.RUNNING ...

Comment (Flanker32) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/function/FunctionNodePresenter.java]: Yes, change to render function node status with SDK result

**Pull request merged**